src/lib.rs

```diff
@@ -112,129 +155,134 @@ pub fn glob_with(pattern: &str, options: &MatchOptions) -> Paths {
     if root.is_some() && check_windows_verbatim(root.as_ref().unwrap()) {
         // FIXME: How do we want to handle verbatim paths? I'm inclined to return nothing,
         // since we can't very well find all UNC shares with a 1-letter server name.
-        return Paths {
+        return Ok(Paths {
             dir_patterns: Vec::new(),
             require_dir: false,
             options: options.clone(),
             todo: Vec::new(),
-        };
+            scope: None,
+        });
     }
 
     let scope = root.map(to_scope).unwrap_or_else(|| Path::new("."));
 
     let dir_patterns = pattern.slice_from(cmp::min(root_len, pattern.len()))
                        .split_terminator(is_sep)
-                       .map(|s| Pattern::new(s))
+                       .map(|s| Pattern::new(s).unwrap())
                        .collect::<Vec<Pattern>>();
 
     let require_dir = pattern.chars().next_back().map(is_sep) == Some(true);
-
     let mut todo = Vec::new();
-    if dir_patterns.len() > 0 {
-        // Shouldn't happen, but we're using -1 as a special index.
-        assert!(dir_patterns.len() < -1 as usize);
-
-        fill_todo(&mut todo, dir_patterns.as_slice(), 0, &scope, options);
-    }
 
-    Paths {
+    Ok(Paths {
         dir_patterns: dir_patterns,
         require_dir: require_dir,
         options: options.clone(),
         todo: todo,
-    }
+        scope: Some(scope),
+    })
+}
+
+/// A glob iteration error.
+///
+/// This is typically returned when a particular path cannot be read
+/// to determine if its contents match the glob pattern. This is possible
+/// if the program lacks the permissions, for example.
+pub struct GlobError {
+  /// The Path that the error corresponds to.
+  pub path: Path,
+
+  /// The error in question.
+  pub error: IoError,
+}
```

Comment:
Sure thing! I try hard to use 4 spaces since I know that's what you guys use but my configuration is to use 2, so a few slip here and there. I'll remember to always triple check from now on so that they're all 4.